### ISSUE TITLE
Change how the edit of the account is performed

### ISSUE THREAD

**Issue by RiccardoM** [desmos-labs/desmos#170]
## Context
I was working on implementing the accounts support inside Mooncake and I realized a bug inside the account module: we currently cannot delete a field. 

If we use [`MsgEditProfile`](https://github.com/desmos-labs/desmos/blob/master/x/profile/internal/types/msgs.go#L106) setting the fields we want delete to either `""` or `nil`, what would happen is that when the message gets handled by the handler ([here](https://github.com/desmos-labs/desmos/blob/master/x/profile/internal/keeper/handler.go#L68)) the `nil` values would be replaced with the existing ones. 

Originally the solution was to ask the user to explicitly tell which fields he wanted to delete by assigning them the `default` value (which would then be reserved), but I think that this makes the working for the clients a lot harder. 

I then realized two things: 

1. Clients 99.99% of the time will show what are the current values to the user before allowing him to edit them.

2. The _Create a profile_ and _Create an account_ terms are usually interchangeable in other social network platforms, but not on ours. We refer as an _account_ when talking about a Cosmos account, and to a _profile_ when we talk about a social network profile. This can be confusing to users (I already got someone asking me the difference). 

## Solution
I've thought that the best solution to both of the problems above is pretty simple: merge the current `MsgCreateProfile` and `MsgEditProfile` into a single `MsgSaveProfile`. 

What this new `MsgSaveProfile` would do is simply __overwrite all the data for the current profile__. That being said: 

- if the developer wants to preserve the current value of a field, he should pass that value inside the message
- if a developer wants to delete the value of a field, he should pass either `""` or `nil` to it. 

This should make things a lot easier for developers and also for users as there is not going to be a _create profile_ term anymore. 

### CLI changes
With this change in mind, the current `tx profile edit` CLI command should change accordingly. What I thought it might be a good implementation is the following: 

```
desmoscli tx profiles edit \
  --name <name> \
  --surname <surname> \ 
  --moniker <moniker> \
 -- ...
 --from jack
``` 

In short, what I envision for this new command is: 

- make all the editable fields __optional__ by putting them as flags
- put the flags default value to `""` 

Once the command is executed, the following things should happen: 

1. Get the currently existing account from the chain. 
2. Replace the value of each field which associated flag has given `""` with the current profile value 
3. Create and send the message

This will be exactly the same method that all clients will use in the future as well.

**Comment by bragaz** [comment]
All clear, do you want to leave the cmd name `edit` or call it `save`?

**Comment by bragaz** [comment]
btw, I can start work on this when I finished post's reports test (ideally tomorrow). Let me know if you need this asap.

**Comment by RiccardoM** [comment]
@bragaz I've added this to the `v0.6.1` intermediary milestone. It's fine to start working on it tomorrow. Ideally I would like to publish that version by the end of the month so we should make it

**Comment by kwunyeung** [comment]
Would the implementation of editing validator description help as a reference? 

https://github.com/cosmos/cosmos-sdk/blob/218ec99508417636ccd47477bc7aad96ac0ac804/x/staking/handler.go#L132

Some similar situation you mentioned in early stage of the validator description.
